// @mirohq/miro-api/dist/model/enterpriseGetOrganization400Response.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.EnterpriseGetOrganization400Response = void 0;
/**
 * @internal
 * Test error object
 */
class EnterpriseGetOrganization400Response {
    /** @ignore */
    static getAttributeTypeMap() {
        return EnterpriseGetOrganization400Response.attributeTypeMap;
    }
}
exports.EnterpriseGetOrganization400Response = EnterpriseGetOrganization400Response;
/** @ignore */
EnterpriseGetOrganization400Response.discriminator = undefined;
/** @ignore */
EnterpriseGetOrganization400Response.attributeTypeMap = [
    {
        name: 'code',
        baseName: 'code',
        type: 'string',
    },
    {
        name: 'message',
        baseName: 'message',
        type: 'string',
    },
    {
        name: 'status',
        baseName: 'status',
        type: 'number',
    },
    {
        name: 'type',
        baseName: 'type',
        type: 'string',
    },
];
