// @mirohq/miro-api/dist/model/frameDataPlatformContainers.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FrameDataPlatformContainers = void 0;
/**
 * @internal
 * Contains frame item data, such as the title, frame type, or frame format. For information about the JSON properties, see [Data](https://developers.miro.com/reference/data).
 */
class FrameDataPlatformContainers {
    constructor() {
        /**
         * Only custom frames are supported at the moment. Possible values: `custom`, `desktop`, `phone`, `tablet`, `a4`, `letter`, `ratio_1x1`, `ratio_4x3`, `ratio_16x9`
         */
        this['format'] = FrameDataPlatformContainers.FormatEnum.Custom;
        /**
         * Only free form frames are supported at the moment. Possible values: `freeform`, `heap`, `grid`, `rows`, `columns`
         */
        this['type'] = FrameDataPlatformContainers.TypeEnum.Freeform;
    }
    /** @ignore */
    static getAttributeTypeMap() {
        return FrameDataPlatformContainers.attributeTypeMap;
    }
}
exports.FrameDataPlatformContainers = FrameDataPlatformContainers;
/** @ignore */
FrameDataPlatformContainers.discriminator = undefined;
/** @ignore */
FrameDataPlatformContainers.attributeTypeMap = [
    {
        name: 'format',
        baseName: 'format',
        type: 'FrameDataPlatformContainers.FormatEnum',
    },
    {
        name: 'title',
        baseName: 'title',
        type: 'string',
    },
    {
        name: 'type',
        baseName: 'type',
        type: 'FrameDataPlatformContainers.TypeEnum',
    },
];
(function (FrameDataPlatformContainers) {
    FrameDataPlatformContainers.FormatEnum = {
        Custom: 'custom',
        Desktop: 'desktop',
        Phone: 'phone',
        Tablet: 'tablet',
        A4: 'a4',
        Letter: 'letter',
        Ratio1x1: 'ratio_1x1',
        Ratio4x3: 'ratio_4x3',
        Ratio16x9: 'ratio_16x9',
    };
    FrameDataPlatformContainers.TypeEnum = {
        Freeform: 'freeform',
        Heap: 'heap',
        Grid: 'grid',
        Rows: 'rows',
        Columns: 'columns',
    };
})(FrameDataPlatformContainers = exports.FrameDataPlatformContainers || (exports.FrameDataPlatformContainers = {}));
