// @mirohq/miro-api/dist/model/models.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObjectSerializer = void 0;
__exportStar(require("./appCardCreateRequest"), exports);
__exportStar(require("./appCardData"), exports);
__exportStar(require("./appCardDataChanges"), exports);
__exportStar(require("./appCardDataPlatformTags"), exports);
__exportStar(require("./appCardItem"), exports);
__exportStar(require("./appCardStyle"), exports);
__exportStar(require("./appCardUpdateRequest"), exports);
__exportStar(require("./board"), exports);
__exportStar(require("./boardChanges"), exports);
__exportStar(require("./boardDataClassificationLabel"), exports);
__exportStar(require("./boardLinks"), exports);
__exportStar(require("./boardMember"), exports);
__exportStar(require("./boardMemberChanges"), exports);
__exportStar(require("./boardMemberWithLinks"), exports);
__exportStar(require("./boardMembersInvite"), exports);
__exportStar(require("./boardMembersPagedResponse"), exports);
__exportStar(require("./boardPermissionsPolicy"), exports);
__exportStar(require("./boardPolicy"), exports);
__exportStar(require("./boardPolicyChange"), exports);
__exportStar(require("./boardSharingPolicy"), exports);
__exportStar(require("./boardSharingPolicyChange"), exports);
__exportStar(require("./boardWithLinks"), exports);
__exportStar(require("./boardsPagedResponse"), exports);
__exportStar(require("./caption"), exports);
__exportStar(require("./cardCreateRequest"), exports);
__exportStar(require("./cardData"), exports);
__exportStar(require("./cardDataPlatformTags"), exports);
__exportStar(require("./cardItem"), exports);
__exportStar(require("./cardStyle"), exports);
__exportStar(require("./cardUpdateRequest"), exports);
__exportStar(require("./connectorChangesData"), exports);
__exportStar(require("./connectorCreationData"), exports);
__exportStar(require("./connectorStyle"), exports);
__exportStar(require("./connectorWithLinks"), exports);
__exportStar(require("./connectorsCursorPaged"), exports);
__exportStar(require("./createTeamRequest"), exports);
__exportStar(require("./createdBy"), exports);
__exportStar(require("./createdByPlatformContainers"), exports);
__exportStar(require("./createdByPlatformTags"), exports);
__exportStar(require("./customField"), exports);
__exportStar(require("./customFieldPlatformTags"), exports);
__exportStar(require("./dataClassificationLabel"), exports);
__exportStar(require("./dataClassificationLabelId"), exports);
__exportStar(require("./dataClassificationOrganizationSettings"), exports);
__exportStar(require("./dataClassificationTeamSettings"), exports);
__exportStar(require("./documentCreateRequest"), exports);
__exportStar(require("./documentData"), exports);
__exportStar(require("./documentDataPlatformTags"), exports);
__exportStar(require("./documentItem"), exports);
__exportStar(require("./documentUpdateRequest"), exports);
__exportStar(require("./documentUrlData"), exports);
__exportStar(require("./documentUrlDataChanges"), exports);
__exportStar(require("./embedCreateRequest"), exports);
__exportStar(require("./embedData"), exports);
__exportStar(require("./embedDataPlatformTags"), exports);
__exportStar(require("./embedItem"), exports);
__exportStar(require("./embedUpdateRequest"), exports);
__exportStar(require("./embedUrlData"), exports);
__exportStar(require("./embedUrlDataChanges"), exports);
__exportStar(require("./enterpriseGetOrganization400Response"), exports);
__exportStar(require("./enterpriseGetOrganizationMembers200Response"), exports);
__exportStar(require("./errorObj"), exports);
__exportStar(require("./fixedRatioGeometry"), exports);
__exportStar(require("./fixedRatioNoRotationGeometry"), exports);
__exportStar(require("./frameChanges"), exports);
__exportStar(require("./frameCreateRequest"), exports);
__exportStar(require("./frameData"), exports);
__exportStar(require("./frameDataPlatformContainers"), exports);
__exportStar(require("./frameDataPlatformTags"), exports);
__exportStar(require("./frameItem"), exports);
__exportStar(require("./frameStyle"), exports);
__exportStar(require("./frameUpdateRequest"), exports);
__exportStar(require("./genericItem"), exports);
__exportStar(require("./genericItemCursorPaged"), exports);
__exportStar(require("./genericItemCursorPagedPlatformContainers"), exports);
__exportStar(require("./genericItemUpdate"), exports);
__exportStar(require("./geometry"), exports);
__exportStar(require("./geometryNoRotation"), exports);
__exportStar(require("./geometryPlatformContainers"), exports);
__exportStar(require("./geometryPlatformTags"), exports);
__exportStar(require("./getBoards400Response"), exports);
__exportStar(require("./getTagsResponse"), exports);
__exportStar(require("./imageCreateRequest"), exports);
__exportStar(require("./imageData"), exports);
__exportStar(require("./imageDataPlatformTags"), exports);
__exportStar(require("./imageItem"), exports);
__exportStar(require("./imageUpdateRequest"), exports);
__exportStar(require("./imageUrlData"), exports);
__exportStar(require("./imageUrlDataChanges"), exports);
__exportStar(require("./invitationError"), exports);
__exportStar(require("./invitationResult"), exports);
__exportStar(require("./itemConnectionChangesData"), exports);
__exportStar(require("./itemConnectionCreationData"), exports);
__exportStar(require("./itemConnectionWithLinks"), exports);
__exportStar(require("./itemPagedResponse"), exports);
__exportStar(require("./modifiedBy"), exports);
__exportStar(require("./modifiedByPlatformContainers"), exports);
__exportStar(require("./modifiedByPlatformTags"), exports);
__exportStar(require("./organization"), exports);
__exportStar(require("./organizationMember"), exports);
__exportStar(require("./organizationMembersSearchByEmailsResponse"), exports);
__exportStar(require("./organizationMembersSearchResponse"), exports);
__exportStar(require("./pageLinks"), exports);
__exportStar(require("./pageLinksPlatformContainers"), exports);
__exportStar(require("./pageLinksPlatformTags"), exports);
__exportStar(require("./parent"), exports);
__exportStar(require("./parentLinksEnvelope"), exports);
__exportStar(require("./parentLinksEnvelopePlatformContainers"), exports);
__exportStar(require("./parentPlatformContainers"), exports);
__exportStar(require("./picture"), exports);
__exportStar(require("./position"), exports);
__exportStar(require("./positionChange"), exports);
__exportStar(require("./positionChangePlatformContainers"), exports);
__exportStar(require("./positionPlatformContainers"), exports);
__exportStar(require("./positionPlatformTags"), exports);
__exportStar(require("./relativeOffset"), exports);
__exportStar(require("./selfLink"), exports);
__exportStar(require("./selfLinkPlatformContainers"), exports);
__exportStar(require("./selfLinkPlatformTags"), exports);
__exportStar(require("./shapeCreateRequest"), exports);
__exportStar(require("./shapeData"), exports);
__exportStar(require("./shapeDataPlatformTags"), exports);
__exportStar(require("./shapeItem"), exports);
__exportStar(require("./shapeStyle"), exports);
__exportStar(require("./shapeUpdateRequest"), exports);
__exportStar(require("./stickyNoteCreateRequest"), exports);
__exportStar(require("./stickyNoteData"), exports);
__exportStar(require("./stickyNoteDataPlatformTags"), exports);
__exportStar(require("./stickyNoteItem"), exports);
__exportStar(require("./stickyNoteStyle"), exports);
__exportStar(require("./stickyNoteUpdateRequest"), exports);
__exportStar(require("./tag"), exports);
__exportStar(require("./tagCreateRequest"), exports);
__exportStar(require("./tagUpdateRequest"), exports);
__exportStar(require("./tagWithLinks"), exports);
__exportStar(require("./tagsPagedResponse"), exports);
__exportStar(require("./team"), exports);
__exportStar(require("./teamAccountDiscoverySettings"), exports);
__exportStar(require("./teamAccountDiscoverySettingsChanges"), exports);
__exportStar(require("./teamChanges"), exports);
__exportStar(require("./teamCollaborationSettings"), exports);
__exportStar(require("./teamCollaborationSettingsChanges"), exports);
__exportStar(require("./teamCopyAccessLevelSettings"), exports);
__exportStar(require("./teamCopyAccessLevelSettingsChanges"), exports);
__exportStar(require("./teamInvitationSettings"), exports);
__exportStar(require("./teamInvitationSettingsChanges"), exports);
__exportStar(require("./teamMember"), exports);
__exportStar(require("./teamMemberChanges"), exports);
__exportStar(require("./teamMemberInvite"), exports);
__exportStar(require("./teamSettings"), exports);
__exportStar(require("./teamSettingsChanges"), exports);
__exportStar(require("./teamSharingPolicySettings"), exports);
__exportStar(require("./teamSharingPolicySettingsChanges"), exports);
__exportStar(require("./textCreateRequest"), exports);
__exportStar(require("./textData"), exports);
__exportStar(require("./textDataPlatformTags"), exports);
__exportStar(require("./textItem"), exports);
__exportStar(require("./textStyle"), exports);
__exportStar(require("./textUpdateRequest"), exports);
__exportStar(require("./updateBoardsDataClassificationLabel"), exports);
__exportStar(require("./updateBoardsDataClassificationLabelRequest"), exports);
__exportStar(require("./updateTeamSettingsRequest"), exports);
__exportStar(require("./userInfoShort"), exports);
__exportStar(require("./widgetData"), exports);
__exportStar(require("./widgetDataPlatformTags"), exports);
__exportStar(require("./widgetLinks"), exports);
__exportStar(require("./widgetLinksPlatformContainers"), exports);
__exportStar(require("./widthOnlyAdjustableGeometry"), exports);
const appCardCreateRequest_1 = require("./appCardCreateRequest");
const appCardData_1 = require("./appCardData");
const appCardDataChanges_1 = require("./appCardDataChanges");
const appCardDataPlatformTags_1 = require("./appCardDataPlatformTags");
const appCardItem_1 = require("./appCardItem");
const appCardStyle_1 = require("./appCardStyle");
const appCardUpdateRequest_1 = require("./appCardUpdateRequest");
const board_1 = require("./board");
const boardChanges_1 = require("./boardChanges");
const boardDataClassificationLabel_1 = require("./boardDataClassificationLabel");
const boardLinks_1 = require("./boardLinks");
const boardMember_1 = require("./boardMember");
const boardMemberChanges_1 = require("./boardMemberChanges");
const boardMemberWithLinks_1 = require("./boardMemberWithLinks");
const boardMembersInvite_1 = require("./boardMembersInvite");
const boardMembersPagedResponse_1 = require("./boardMembersPagedResponse");
const boardPermissionsPolicy_1 = require("./boardPermissionsPolicy");
const boardPolicy_1 = require("./boardPolicy");
const boardPolicyChange_1 = require("./boardPolicyChange");
const boardSharingPolicy_1 = require("./boardSharingPolicy");
const boardSharingPolicyChange_1 = require("./boardSharingPolicyChange");
const boardWithLinks_1 = require("./boardWithLinks");
const boardsPagedResponse_1 = require("./boardsPagedResponse");
const caption_1 = require("./caption");
const cardCreateRequest_1 = require("./cardCreateRequest");
const cardData_1 = require("./cardData");
const cardDataPlatformTags_1 = require("./cardDataPlatformTags");
const cardItem_1 = require("./cardItem");
const cardStyle_1 = require("./cardStyle");
const cardUpdateRequest_1 = require("./cardUpdateRequest");
const connectorChangesData_1 = require("./connectorChangesData");
const connectorCreationData_1 = require("./connectorCreationData");
const connectorStyle_1 = require("./connectorStyle");
const connectorWithLinks_1 = require("./connectorWithLinks");
const connectorsCursorPaged_1 = require("./connectorsCursorPaged");
const createTeamRequest_1 = require("./createTeamRequest");
const createdBy_1 = require("./createdBy");
const createdByPlatformContainers_1 = require("./createdByPlatformContainers");
const createdByPlatformTags_1 = require("./createdByPlatformTags");
const customField_1 = require("./customField");
const customFieldPlatformTags_1 = require("./customFieldPlatformTags");
const dataClassificationLabel_1 = require("./dataClassificationLabel");
const dataClassificationLabelId_1 = require("./dataClassificationLabelId");
const dataClassificationOrganizationSettings_1 = require("./dataClassificationOrganizationSettings");
const dataClassificationTeamSettings_1 = require("./dataClassificationTeamSettings");
const documentCreateRequest_1 = require("./documentCreateRequest");
const documentData_1 = require("./documentData");
const documentDataPlatformTags_1 = require("./documentDataPlatformTags");
const documentItem_1 = require("./documentItem");
const documentUpdateRequest_1 = require("./documentUpdateRequest");
const documentUrlData_1 = require("./documentUrlData");
const documentUrlDataChanges_1 = require("./documentUrlDataChanges");
const embedCreateRequest_1 = require("./embedCreateRequest");
const embedData_1 = require("./embedData");
const embedDataPlatformTags_1 = require("./embedDataPlatformTags");
const embedItem_1 = require("./embedItem");
const embedUpdateRequest_1 = require("./embedUpdateRequest");
const embedUrlData_1 = require("./embedUrlData");
const embedUrlDataChanges_1 = require("./embedUrlDataChanges");
const enterpriseGetOrganization400Response_1 = require("./enterpriseGetOrganization400Response");
const enterpriseGetOrganizationMembers200Response_1 = require("./enterpriseGetOrganizationMembers200Response");
const errorObj_1 = require("./errorObj");
const fixedRatioGeometry_1 = require("./fixedRatioGeometry");
const fixedRatioNoRotationGeometry_1 = require("./fixedRatioNoRotationGeometry");
const frameChanges_1 = require("./frameChanges");
const frameCreateRequest_1 = require("./frameCreateRequest");
const frameData_1 = require("./frameData");
const frameDataPlatformContainers_1 = require("./frameDataPlatformContainers");
const frameDataPlatformTags_1 = require("./frameDataPlatformTags");
const frameItem_1 = require("./frameItem");
const frameStyle_1 = require("./frameStyle");
const frameUpdateRequest_1 = require("./frameUpdateRequest");
const genericItem_1 = require("./genericItem");
const genericItemCursorPaged_1 = require("./genericItemCursorPaged");
const genericItemCursorPagedPlatformContainers_1 = require("./genericItemCursorPagedPlatformContainers");
const genericItemUpdate_1 = require("./genericItemUpdate");
const geometry_1 = require("./geometry");
const geometryNoRotation_1 = require("./geometryNoRotation");
const geometryPlatformContainers_1 = require("./geometryPlatformContainers");
const geometryPlatformTags_1 = require("./geometryPlatformTags");
const getBoards400Response_1 = require("./getBoards400Response");
const getTagsResponse_1 = require("./getTagsResponse");
const imageCreateRequest_1 = require("./imageCreateRequest");
const imageData_1 = require("./imageData");
const imageDataPlatformTags_1 = require("./imageDataPlatformTags");
const imageItem_1 = require("./imageItem");
const imageUpdateRequest_1 = require("./imageUpdateRequest");
const imageUrlData_1 = require("./imageUrlData");
const imageUrlDataChanges_1 = require("./imageUrlDataChanges");
const invitationError_1 = require("./invitationError");
const invitationResult_1 = require("./invitationResult");
const itemConnectionChangesData_1 = require("./itemConnectionChangesData");
const itemConnectionCreationData_1 = require("./itemConnectionCreationData");
const itemConnectionWithLinks_1 = require("./itemConnectionWithLinks");
const itemPagedResponse_1 = require("./itemPagedResponse");
const modifiedBy_1 = require("./modifiedBy");
const modifiedByPlatformContainers_1 = require("./modifiedByPlatformContainers");
const modifiedByPlatformTags_1 = require("./modifiedByPlatformTags");
const organization_1 = require("./organization");
const organizationMember_1 = require("./organizationMember");
const organizationMembersSearchByEmailsResponse_1 = require("./organizationMembersSearchByEmailsResponse");
const organizationMembersSearchResponse_1 = require("./organizationMembersSearchResponse");
const pageLinks_1 = require("./pageLinks");
const pageLinksPlatformContainers_1 = require("./pageLinksPlatformContainers");
const pageLinksPlatformTags_1 = require("./pageLinksPlatformTags");
const parent_1 = require("./parent");
const parentLinksEnvelope_1 = require("./parentLinksEnvelope");
const parentLinksEnvelopePlatformContainers_1 = require("./parentLinksEnvelopePlatformContainers");
const parentPlatformContainers_1 = require("./parentPlatformContainers");
const picture_1 = require("./picture");
const position_1 = require("./position");
const positionChange_1 = require("./positionChange");
const positionChangePlatformContainers_1 = require("./positionChangePlatformContainers");
const positionPlatformContainers_1 = require("./positionPlatformContainers");
const positionPlatformTags_1 = require("./positionPlatformTags");
const relativeOffset_1 = require("./relativeOffset");
const selfLink_1 = require("./selfLink");
const selfLinkPlatformContainers_1 = require("./selfLinkPlatformContainers");
const selfLinkPlatformTags_1 = require("./selfLinkPlatformTags");
const shapeCreateRequest_1 = require("./shapeCreateRequest");
const shapeData_1 = require("./shapeData");
const shapeDataPlatformTags_1 = require("./shapeDataPlatformTags");
const shapeItem_1 = require("./shapeItem");
const shapeStyle_1 = require("./shapeStyle");
const shapeUpdateRequest_1 = require("./shapeUpdateRequest");
const stickyNoteCreateRequest_1 = require("./stickyNoteCreateRequest");
const stickyNoteData_1 = require("./stickyNoteData");
const stickyNoteDataPlatformTags_1 = require("./stickyNoteDataPlatformTags");
const stickyNoteItem_1 = require("./stickyNoteItem");
const stickyNoteStyle_1 = require("./stickyNoteStyle");
const stickyNoteUpdateRequest_1 = require("./stickyNoteUpdateRequest");
const tag_1 = require("./tag");
const tagCreateRequest_1 = require("./tagCreateRequest");
const tagUpdateRequest_1 = require("./tagUpdateRequest");
const tagWithLinks_1 = require("./tagWithLinks");
const tagsPagedResponse_1 = require("./tagsPagedResponse");
const team_1 = require("./team");
const teamAccountDiscoverySettings_1 = require("./teamAccountDiscoverySettings");
const teamAccountDiscoverySettingsChanges_1 = require("./teamAccountDiscoverySettingsChanges");
const teamChanges_1 = require("./teamChanges");
const teamCollaborationSettings_1 = require("./teamCollaborationSettings");
const teamCollaborationSettingsChanges_1 = require("./teamCollaborationSettingsChanges");
const teamCopyAccessLevelSettings_1 = require("./teamCopyAccessLevelSettings");
const teamCopyAccessLevelSettingsChanges_1 = require("./teamCopyAccessLevelSettingsChanges");
const teamInvitationSettings_1 = require("./teamInvitationSettings");
const teamInvitationSettingsChanges_1 = require("./teamInvitationSettingsChanges");
const teamMember_1 = require("./teamMember");
const teamMemberChanges_1 = require("./teamMemberChanges");
const teamMemberInvite_1 = require("./teamMemberInvite");
const teamSettings_1 = require("./teamSettings");
const teamSettingsChanges_1 = require("./teamSettingsChanges");
const teamSharingPolicySettings_1 = require("./teamSharingPolicySettings");
const teamSharingPolicySettingsChanges_1 = require("./teamSharingPolicySettingsChanges");
const textCreateRequest_1 = require("./textCreateRequest");
const textData_1 = require("./textData");
const textDataPlatformTags_1 = require("./textDataPlatformTags");
const textItem_1 = require("./textItem");
const textStyle_1 = require("./textStyle");
const textUpdateRequest_1 = require("./textUpdateRequest");
const updateBoardsDataClassificationLabel_1 = require("./updateBoardsDataClassificationLabel");
const updateBoardsDataClassificationLabelRequest_1 = require("./updateBoardsDataClassificationLabelRequest");
const updateTeamSettingsRequest_1 = require("./updateTeamSettingsRequest");
const userInfoShort_1 = require("./userInfoShort");
const widgetData_1 = require("./widgetData");
const widgetDataPlatformTags_1 = require("./widgetDataPlatformTags");
const widgetLinks_1 = require("./widgetLinks");
const widgetLinksPlatformContainers_1 = require("./widgetLinksPlatformContainers");
const widthOnlyAdjustableGeometry_1 = require("./widthOnlyAdjustableGeometry");
/* tslint:disable:no-unused-variable */
let primitives = ['string', 'boolean', 'double', 'integer', 'long', 'float', 'number', 'any'];
let enumsMap = {
    'AppCardData.StatusEnum': appCardData_1.AppCardData.StatusEnum,
    'AppCardDataChanges.StatusEnum': appCardDataChanges_1.AppCardDataChanges.StatusEnum,
    'AppCardDataPlatformTags.StatusEnum': appCardDataPlatformTags_1.AppCardDataPlatformTags.StatusEnum,
    'BoardMember.RoleEnum': boardMember_1.BoardMember.RoleEnum,
    'BoardMemberChanges.RoleEnum': boardMemberChanges_1.BoardMemberChanges.RoleEnum,
    'BoardMemberWithLinks.RoleEnum': boardMemberWithLinks_1.BoardMemberWithLinks.RoleEnum,
    'BoardMembersInvite.RoleEnum': boardMembersInvite_1.BoardMembersInvite.RoleEnum,
    'BoardPermissionsPolicy.CollaborationToolsStartAccessEnum': boardPermissionsPolicy_1.BoardPermissionsPolicy.CollaborationToolsStartAccessEnum,
    'BoardPermissionsPolicy.CopyAccessEnum': boardPermissionsPolicy_1.BoardPermissionsPolicy.CopyAccessEnum,
    'BoardPermissionsPolicy.SharingAccessEnum': boardPermissionsPolicy_1.BoardPermissionsPolicy.SharingAccessEnum,
    'BoardSharingPolicy.AccessEnum': boardSharingPolicy_1.BoardSharingPolicy.AccessEnum,
    'BoardSharingPolicy.InviteToAccountAndBoardLinkAccessEnum': boardSharingPolicy_1.BoardSharingPolicy.InviteToAccountAndBoardLinkAccessEnum,
    'BoardSharingPolicy.OrganizationAccessEnum': boardSharingPolicy_1.BoardSharingPolicy.OrganizationAccessEnum,
    'BoardSharingPolicy.TeamAccessEnum': boardSharingPolicy_1.BoardSharingPolicy.TeamAccessEnum,
    'BoardSharingPolicyChange.AccessEnum': boardSharingPolicyChange_1.BoardSharingPolicyChange.AccessEnum,
    'BoardSharingPolicyChange.InviteToAccountAndBoardLinkAccessEnum': boardSharingPolicyChange_1.BoardSharingPolicyChange.InviteToAccountAndBoardLinkAccessEnum,
    'BoardSharingPolicyChange.OrganizationAccessEnum': boardSharingPolicyChange_1.BoardSharingPolicyChange.OrganizationAccessEnum,
    'BoardSharingPolicyChange.TeamAccessEnum': boardSharingPolicyChange_1.BoardSharingPolicyChange.TeamAccessEnum,
    'Caption.TextAlignVerticalEnum': caption_1.Caption.TextAlignVerticalEnum,
    'ConnectorChangesData.ShapeEnum': connectorChangesData_1.ConnectorChangesData.ShapeEnum,
    'ConnectorCreationData.ShapeEnum': connectorCreationData_1.ConnectorCreationData.ShapeEnum,
    'ConnectorStyle.EndStrokeCapEnum': connectorStyle_1.ConnectorStyle.EndStrokeCapEnum,
    'ConnectorStyle.StartStrokeCapEnum': connectorStyle_1.ConnectorStyle.StartStrokeCapEnum,
    'ConnectorStyle.StrokeStyleEnum': connectorStyle_1.ConnectorStyle.StrokeStyleEnum,
    'ConnectorStyle.TextOrientationEnum': connectorStyle_1.ConnectorStyle.TextOrientationEnum,
    'ConnectorWithLinks.ShapeEnum': connectorWithLinks_1.ConnectorWithLinks.ShapeEnum,
    'CustomField.IconShapeEnum': customField_1.CustomField.IconShapeEnum,
    'CustomFieldPlatformTags.IconShapeEnum': customFieldPlatformTags_1.CustomFieldPlatformTags.IconShapeEnum,
    'EmbedData.ModeEnum': embedData_1.EmbedData.ModeEnum,
    'EmbedDataPlatformTags.ModeEnum': embedDataPlatformTags_1.EmbedDataPlatformTags.ModeEnum,
    'EmbedUrlData.ModeEnum': embedUrlData_1.EmbedUrlData.ModeEnum,
    'EmbedUrlDataChanges.ModeEnum': embedUrlDataChanges_1.EmbedUrlDataChanges.ModeEnum,
    'FrameChanges.FormatEnum': frameChanges_1.FrameChanges.FormatEnum,
    'FrameChanges.TypeEnum': frameChanges_1.FrameChanges.TypeEnum,
    'FrameData.FormatEnum': frameData_1.FrameData.FormatEnum,
    'FrameData.TypeEnum': frameData_1.FrameData.TypeEnum,
    'FrameDataPlatformContainers.FormatEnum': frameDataPlatformContainers_1.FrameDataPlatformContainers.FormatEnum,
    'FrameDataPlatformContainers.TypeEnum': frameDataPlatformContainers_1.FrameDataPlatformContainers.TypeEnum,
    'FrameDataPlatformTags.FormatEnum': frameDataPlatformTags_1.FrameDataPlatformTags.FormatEnum,
    'FrameDataPlatformTags.TypeEnum': frameDataPlatformTags_1.FrameDataPlatformTags.TypeEnum,
    'ItemConnectionChangesData.SnapToEnum': itemConnectionChangesData_1.ItemConnectionChangesData.SnapToEnum,
    'ItemConnectionCreationData.SnapToEnum': itemConnectionCreationData_1.ItemConnectionCreationData.SnapToEnum,
    'Organization.PlanEnum': organization_1.Organization.PlanEnum,
    'OrganizationMember.LicenseEnum': organizationMember_1.OrganizationMember.LicenseEnum,
    'OrganizationMember.RoleEnum': organizationMember_1.OrganizationMember.RoleEnum,
    'Position.OriginEnum': position_1.Position.OriginEnum,
    'Position.RelativeToEnum': position_1.Position.RelativeToEnum,
    'PositionChange.OriginEnum': positionChange_1.PositionChange.OriginEnum,
    'PositionChangePlatformContainers.OriginEnum': positionChangePlatformContainers_1.PositionChangePlatformContainers.OriginEnum,
    'PositionPlatformContainers.OriginEnum': positionPlatformContainers_1.PositionPlatformContainers.OriginEnum,
    'PositionPlatformContainers.RelativeToEnum': positionPlatformContainers_1.PositionPlatformContainers.RelativeToEnum,
    'PositionPlatformTags.OriginEnum': positionPlatformTags_1.PositionPlatformTags.OriginEnum,
    'PositionPlatformTags.RelativeToEnum': positionPlatformTags_1.PositionPlatformTags.RelativeToEnum,
    'ShapeData.ShapeEnum': shapeData_1.ShapeData.ShapeEnum,
    'ShapeDataPlatformTags.ShapeEnum': shapeDataPlatformTags_1.ShapeDataPlatformTags.ShapeEnum,
    'ShapeStyle.BorderStyleEnum': shapeStyle_1.ShapeStyle.BorderStyleEnum,
    'ShapeStyle.FontFamilyEnum': shapeStyle_1.ShapeStyle.FontFamilyEnum,
    'ShapeStyle.TextAlignEnum': shapeStyle_1.ShapeStyle.TextAlignEnum,
    'ShapeStyle.TextAlignVerticalEnum': shapeStyle_1.ShapeStyle.TextAlignVerticalEnum,
    'StickyNoteData.ShapeEnum': stickyNoteData_1.StickyNoteData.ShapeEnum,
    'StickyNoteDataPlatformTags.ShapeEnum': stickyNoteDataPlatformTags_1.StickyNoteDataPlatformTags.ShapeEnum,
    'StickyNoteStyle.FillColorEnum': stickyNoteStyle_1.StickyNoteStyle.FillColorEnum,
    'StickyNoteStyle.TextAlignEnum': stickyNoteStyle_1.StickyNoteStyle.TextAlignEnum,
    'StickyNoteStyle.TextAlignVerticalEnum': stickyNoteStyle_1.StickyNoteStyle.TextAlignVerticalEnum,
    'Tag.FillColorEnum': tag_1.Tag.FillColorEnum,
    'TagCreateRequest.FillColorEnum': tagCreateRequest_1.TagCreateRequest.FillColorEnum,
    'TagUpdateRequest.FillColorEnum': tagUpdateRequest_1.TagUpdateRequest.FillColorEnum,
    'TagWithLinks.FillColorEnum': tagWithLinks_1.TagWithLinks.FillColorEnum,
    'TeamAccountDiscoverySettings.AccountDiscoveryEnum': teamAccountDiscoverySettings_1.TeamAccountDiscoverySettings.AccountDiscoveryEnum,
    'TeamAccountDiscoverySettingsChanges.AccountDiscoveryEnum': teamAccountDiscoverySettingsChanges_1.TeamAccountDiscoverySettingsChanges.AccountDiscoveryEnum,
    'TeamCollaborationSettings.CoOwnerRoleEnum': teamCollaborationSettings_1.TeamCollaborationSettings.CoOwnerRoleEnum,
    'TeamCollaborationSettingsChanges.CoOwnerRoleEnum': teamCollaborationSettingsChanges_1.TeamCollaborationSettingsChanges.CoOwnerRoleEnum,
    'TeamCopyAccessLevelSettings.CopyAccessLevelEnum': teamCopyAccessLevelSettings_1.TeamCopyAccessLevelSettings.CopyAccessLevelEnum,
    'TeamCopyAccessLevelSettings.CopyAccessLevelLimitationEnum': teamCopyAccessLevelSettings_1.TeamCopyAccessLevelSettings.CopyAccessLevelLimitationEnum,
    'TeamCopyAccessLevelSettingsChanges.CopyAccessLevelEnum': teamCopyAccessLevelSettingsChanges_1.TeamCopyAccessLevelSettingsChanges.CopyAccessLevelEnum,
    'TeamCopyAccessLevelSettingsChanges.CopyAccessLevelLimitationEnum': teamCopyAccessLevelSettingsChanges_1.TeamCopyAccessLevelSettingsChanges.CopyAccessLevelLimitationEnum,
    'TeamInvitationSettings.InviteExternalUsersEnum': teamInvitationSettings_1.TeamInvitationSettings.InviteExternalUsersEnum,
    'TeamInvitationSettings.WhoCanInviteEnum': teamInvitationSettings_1.TeamInvitationSettings.WhoCanInviteEnum,
    'TeamInvitationSettingsChanges.InviteExternalUsersEnum': teamInvitationSettingsChanges_1.TeamInvitationSettingsChanges.InviteExternalUsersEnum,
    'TeamInvitationSettingsChanges.WhoCanInviteEnum': teamInvitationSettingsChanges_1.TeamInvitationSettingsChanges.WhoCanInviteEnum,
    'TeamMember.UserRoleEnum': teamMember_1.TeamMember.UserRoleEnum,
    'TeamSharingPolicySettings.CreateAssetAccessLevelEnum': teamSharingPolicySettings_1.TeamSharingPolicySettings.CreateAssetAccessLevelEnum,
    'TeamSharingPolicySettings.DefaultBoardAccessEnum': teamSharingPolicySettings_1.TeamSharingPolicySettings.DefaultBoardAccessEnum,
    'TeamSharingPolicySettings.DefaultOrganizationAccessEnum': teamSharingPolicySettings_1.TeamSharingPolicySettings.DefaultOrganizationAccessEnum,
    'TeamSharingPolicySettings.DefaultProjectAccessEnum': teamSharingPolicySettings_1.TeamSharingPolicySettings.DefaultProjectAccessEnum,
    'TeamSharingPolicySettings.MoveBoardToAccountEnum': teamSharingPolicySettings_1.TeamSharingPolicySettings.MoveBoardToAccountEnum,
    'TeamSharingPolicySettings.RestrictAllowedDomainsEnum': teamSharingPolicySettings_1.TeamSharingPolicySettings.RestrictAllowedDomainsEnum,
    'TeamSharingPolicySettings.SharingOnAccountEnum': teamSharingPolicySettings_1.TeamSharingPolicySettings.SharingOnAccountEnum,
    'TeamSharingPolicySettings.SharingOnOrganizationEnum': teamSharingPolicySettings_1.TeamSharingPolicySettings.SharingOnOrganizationEnum,
    'TeamSharingPolicySettings.SharingViaPublicLinkEnum': teamSharingPolicySettings_1.TeamSharingPolicySettings.SharingViaPublicLinkEnum,
    'TeamSharingPolicySettingsChanges.CreateAssetAccessLevelEnum': teamSharingPolicySettingsChanges_1.TeamSharingPolicySettingsChanges.CreateAssetAccessLevelEnum,
    'TeamSharingPolicySettingsChanges.DefaultBoardAccessEnum': teamSharingPolicySettingsChanges_1.TeamSharingPolicySettingsChanges.DefaultBoardAccessEnum,
    'TeamSharingPolicySettingsChanges.DefaultOrganizationAccessEnum': teamSharingPolicySettingsChanges_1.TeamSharingPolicySettingsChanges.DefaultOrganizationAccessEnum,
    'TeamSharingPolicySettingsChanges.DefaultProjectAccessEnum': teamSharingPolicySettingsChanges_1.TeamSharingPolicySettingsChanges.DefaultProjectAccessEnum,
    'TeamSharingPolicySettingsChanges.MoveBoardToAccountEnum': teamSharingPolicySettingsChanges_1.TeamSharingPolicySettingsChanges.MoveBoardToAccountEnum,
    'TeamSharingPolicySettingsChanges.RestrictAllowedDomainsEnum': teamSharingPolicySettingsChanges_1.TeamSharingPolicySettingsChanges.RestrictAllowedDomainsEnum,
    'TeamSharingPolicySettingsChanges.SharingOnAccountEnum': teamSharingPolicySettingsChanges_1.TeamSharingPolicySettingsChanges.SharingOnAccountEnum,
    'TeamSharingPolicySettingsChanges.SharingOnOrganizationEnum': teamSharingPolicySettingsChanges_1.TeamSharingPolicySettingsChanges.SharingOnOrganizationEnum,
    'TeamSharingPolicySettingsChanges.SharingViaPublicLinkEnum': teamSharingPolicySettingsChanges_1.TeamSharingPolicySettingsChanges.SharingViaPublicLinkEnum,
    'TextStyle.FontFamilyEnum': textStyle_1.TextStyle.FontFamilyEnum,
    'TextStyle.TextAlignEnum': textStyle_1.TextStyle.TextAlignEnum,
    'WidgetData.ModeEnum': widgetData_1.WidgetData.ModeEnum,
    'WidgetData.StatusEnum': widgetData_1.WidgetData.StatusEnum,
    'WidgetData.ShapeEnum': widgetData_1.WidgetData.ShapeEnum,
    'WidgetData.FormatEnum': widgetData_1.WidgetData.FormatEnum,
    'WidgetData.TypeEnum': widgetData_1.WidgetData.TypeEnum,
    'WidgetDataPlatformTags.ModeEnum': widgetDataPlatformTags_1.WidgetDataPlatformTags.ModeEnum,
    'WidgetDataPlatformTags.StatusEnum': widgetDataPlatformTags_1.WidgetDataPlatformTags.StatusEnum,
    'WidgetDataPlatformTags.ShapeEnum': widgetDataPlatformTags_1.WidgetDataPlatformTags.ShapeEnum,
    'WidgetDataPlatformTags.FormatEnum': widgetDataPlatformTags_1.WidgetDataPlatformTags.FormatEnum,
    'WidgetDataPlatformTags.TypeEnum': widgetDataPlatformTags_1.WidgetDataPlatformTags.TypeEnum,
};
let typeMap = {
    AppCardCreateRequest: appCardCreateRequest_1.AppCardCreateRequest,
    AppCardData: appCardData_1.AppCardData,
    AppCardDataChanges: appCardDataChanges_1.AppCardDataChanges,
    AppCardDataPlatformTags: appCardDataPlatformTags_1.AppCardDataPlatformTags,
    AppCardItem: appCardItem_1.AppCardItem,
    AppCardStyle: appCardStyle_1.AppCardStyle,
    AppCardUpdateRequest: appCardUpdateRequest_1.AppCardUpdateRequest,
    Board: board_1.Board,
    BoardChanges: boardChanges_1.BoardChanges,
    BoardDataClassificationLabel: boardDataClassificationLabel_1.BoardDataClassificationLabel,
    BoardLinks: boardLinks_1.BoardLinks,
    BoardMember: boardMember_1.BoardMember,
    BoardMemberChanges: boardMemberChanges_1.BoardMemberChanges,
    BoardMemberWithLinks: boardMemberWithLinks_1.BoardMemberWithLinks,
    BoardMembersInvite: boardMembersInvite_1.BoardMembersInvite,
    BoardMembersPagedResponse: boardMembersPagedResponse_1.BoardMembersPagedResponse,
    BoardPermissionsPolicy: boardPermissionsPolicy_1.BoardPermissionsPolicy,
    BoardPolicy: boardPolicy_1.BoardPolicy,
    BoardPolicyChange: boardPolicyChange_1.BoardPolicyChange,
    BoardSharingPolicy: boardSharingPolicy_1.BoardSharingPolicy,
    BoardSharingPolicyChange: boardSharingPolicyChange_1.BoardSharingPolicyChange,
    BoardWithLinks: boardWithLinks_1.BoardWithLinks,
    BoardsPagedResponse: boardsPagedResponse_1.BoardsPagedResponse,
    Caption: caption_1.Caption,
    CardCreateRequest: cardCreateRequest_1.CardCreateRequest,
    CardData: cardData_1.CardData,
    CardDataPlatformTags: cardDataPlatformTags_1.CardDataPlatformTags,
    CardItem: cardItem_1.CardItem,
    CardStyle: cardStyle_1.CardStyle,
    CardUpdateRequest: cardUpdateRequest_1.CardUpdateRequest,
    ConnectorChangesData: connectorChangesData_1.ConnectorChangesData,
    ConnectorCreationData: connectorCreationData_1.ConnectorCreationData,
    ConnectorStyle: connectorStyle_1.ConnectorStyle,
    ConnectorWithLinks: connectorWithLinks_1.ConnectorWithLinks,
    ConnectorsCursorPaged: connectorsCursorPaged_1.ConnectorsCursorPaged,
    CreateTeamRequest: createTeamRequest_1.CreateTeamRequest,
    CreatedBy: createdBy_1.CreatedBy,
    CreatedByPlatformContainers: createdByPlatformContainers_1.CreatedByPlatformContainers,
    CreatedByPlatformTags: createdByPlatformTags_1.CreatedByPlatformTags,
    CustomField: customField_1.CustomField,
    CustomFieldPlatformTags: customFieldPlatformTags_1.CustomFieldPlatformTags,
    DataClassificationLabel: dataClassificationLabel_1.DataClassificationLabel,
    DataClassificationLabelId: dataClassificationLabelId_1.DataClassificationLabelId,
    DataClassificationOrganizationSettings: dataClassificationOrganizationSettings_1.DataClassificationOrganizationSettings,
    DataClassificationTeamSettings: dataClassificationTeamSettings_1.DataClassificationTeamSettings,
    DocumentCreateRequest: documentCreateRequest_1.DocumentCreateRequest,
    DocumentData: documentData_1.DocumentData,
    DocumentDataPlatformTags: documentDataPlatformTags_1.DocumentDataPlatformTags,
    DocumentItem: documentItem_1.DocumentItem,
    DocumentUpdateRequest: documentUpdateRequest_1.DocumentUpdateRequest,
    DocumentUrlData: documentUrlData_1.DocumentUrlData,
    DocumentUrlDataChanges: documentUrlDataChanges_1.DocumentUrlDataChanges,
    EmbedCreateRequest: embedCreateRequest_1.EmbedCreateRequest,
    EmbedData: embedData_1.EmbedData,
    EmbedDataPlatformTags: embedDataPlatformTags_1.EmbedDataPlatformTags,
    EmbedItem: embedItem_1.EmbedItem,
    EmbedUpdateRequest: embedUpdateRequest_1.EmbedUpdateRequest,
    EmbedUrlData: embedUrlData_1.EmbedUrlData,
    EmbedUrlDataChanges: embedUrlDataChanges_1.EmbedUrlDataChanges,
    EnterpriseGetOrganization400Response: enterpriseGetOrganization400Response_1.EnterpriseGetOrganization400Response,
    EnterpriseGetOrganizationMembers200Response: enterpriseGetOrganizationMembers200Response_1.EnterpriseGetOrganizationMembers200Response,
    ErrorObj: errorObj_1.ErrorObj,
    FixedRatioGeometry: fixedRatioGeometry_1.FixedRatioGeometry,
    FixedRatioNoRotationGeometry: fixedRatioNoRotationGeometry_1.FixedRatioNoRotationGeometry,
    FrameChanges: frameChanges_1.FrameChanges,
    FrameCreateRequest: frameCreateRequest_1.FrameCreateRequest,
    FrameData: frameData_1.FrameData,
    FrameDataPlatformContainers: frameDataPlatformContainers_1.FrameDataPlatformContainers,
    FrameDataPlatformTags: frameDataPlatformTags_1.FrameDataPlatformTags,
    FrameItem: frameItem_1.FrameItem,
    FrameStyle: frameStyle_1.FrameStyle,
    FrameUpdateRequest: frameUpdateRequest_1.FrameUpdateRequest,
    GenericItem: genericItem_1.GenericItem,
    GenericItemCursorPaged: genericItemCursorPaged_1.GenericItemCursorPaged,
    GenericItemCursorPagedPlatformContainers: genericItemCursorPagedPlatformContainers_1.GenericItemCursorPagedPlatformContainers,
    GenericItemUpdate: genericItemUpdate_1.GenericItemUpdate,
    Geometry: geometry_1.Geometry,
    GeometryNoRotation: geometryNoRotation_1.GeometryNoRotation,
    GeometryPlatformContainers: geometryPlatformContainers_1.GeometryPlatformContainers,
    GeometryPlatformTags: geometryPlatformTags_1.GeometryPlatformTags,
    GetBoards400Response: getBoards400Response_1.GetBoards400Response,
    GetTagsResponse: getTagsResponse_1.GetTagsResponse,
    ImageCreateRequest: imageCreateRequest_1.ImageCreateRequest,
    ImageData: imageData_1.ImageData,
    ImageDataPlatformTags: imageDataPlatformTags_1.ImageDataPlatformTags,
    ImageItem: imageItem_1.ImageItem,
    ImageUpdateRequest: imageUpdateRequest_1.ImageUpdateRequest,
    ImageUrlData: imageUrlData_1.ImageUrlData,
    ImageUrlDataChanges: imageUrlDataChanges_1.ImageUrlDataChanges,
    InvitationError: invitationError_1.InvitationError,
    InvitationResult: invitationResult_1.InvitationResult,
    ItemConnectionChangesData: itemConnectionChangesData_1.ItemConnectionChangesData,
    ItemConnectionCreationData: itemConnectionCreationData_1.ItemConnectionCreationData,
    ItemConnectionWithLinks: itemConnectionWithLinks_1.ItemConnectionWithLinks,
    ItemPagedResponse: itemPagedResponse_1.ItemPagedResponse,
    ModifiedBy: modifiedBy_1.ModifiedBy,
    ModifiedByPlatformContainers: modifiedByPlatformContainers_1.ModifiedByPlatformContainers,
    ModifiedByPlatformTags: modifiedByPlatformTags_1.ModifiedByPlatformTags,
    Organization: organization_1.Organization,
    OrganizationMember: organizationMember_1.OrganizationMember,
    OrganizationMembersSearchByEmailsResponse: organizationMembersSearchByEmailsResponse_1.OrganizationMembersSearchByEmailsResponse,
    OrganizationMembersSearchResponse: organizationMembersSearchResponse_1.OrganizationMembersSearchResponse,
    PageLinks: pageLinks_1.PageLinks,
    PageLinksPlatformContainers: pageLinksPlatformContainers_1.PageLinksPlatformContainers,
    PageLinksPlatformTags: pageLinksPlatformTags_1.PageLinksPlatformTags,
    Parent: parent_1.Parent,
    ParentLinksEnvelope: parentLinksEnvelope_1.ParentLinksEnvelope,
    ParentLinksEnvelopePlatformContainers: parentLinksEnvelopePlatformContainers_1.ParentLinksEnvelopePlatformContainers,
    ParentPlatformContainers: parentPlatformContainers_1.ParentPlatformContainers,
    Picture: picture_1.Picture,
    Position: position_1.Position,
    PositionChange: positionChange_1.PositionChange,
    PositionChangePlatformContainers: positionChangePlatformContainers_1.PositionChangePlatformContainers,
    PositionPlatformContainers: positionPlatformContainers_1.PositionPlatformContainers,
    PositionPlatformTags: positionPlatformTags_1.PositionPlatformTags,
    RelativeOffset: relativeOffset_1.RelativeOffset,
    SelfLink: selfLink_1.SelfLink,
    SelfLinkPlatformContainers: selfLinkPlatformContainers_1.SelfLinkPlatformContainers,
    SelfLinkPlatformTags: selfLinkPlatformTags_1.SelfLinkPlatformTags,
    ShapeCreateRequest: shapeCreateRequest_1.ShapeCreateRequest,
    ShapeData: shapeData_1.ShapeData,
    ShapeDataPlatformTags: shapeDataPlatformTags_1.ShapeDataPlatformTags,
    ShapeItem: shapeItem_1.ShapeItem,
    ShapeStyle: shapeStyle_1.ShapeStyle,
    ShapeUpdateRequest: shapeUpdateRequest_1.ShapeUpdateRequest,
    StickyNoteCreateRequest: stickyNoteCreateRequest_1.StickyNoteCreateRequest,
    StickyNoteData: stickyNoteData_1.StickyNoteData,
    StickyNoteDataPlatformTags: stickyNoteDataPlatformTags_1.StickyNoteDataPlatformTags,
    StickyNoteItem: stickyNoteItem_1.StickyNoteItem,
    StickyNoteStyle: stickyNoteStyle_1.StickyNoteStyle,
    StickyNoteUpdateRequest: stickyNoteUpdateRequest_1.StickyNoteUpdateRequest,
    Tag: tag_1.Tag,
    TagCreateRequest: tagCreateRequest_1.TagCreateRequest,
    TagUpdateRequest: tagUpdateRequest_1.TagUpdateRequest,
    TagWithLinks: tagWithLinks_1.TagWithLinks,
    TagsPagedResponse: tagsPagedResponse_1.TagsPagedResponse,
    Team: team_1.Team,
    TeamAccountDiscoverySettings: teamAccountDiscoverySettings_1.TeamAccountDiscoverySettings,
    TeamAccountDiscoverySettingsChanges: teamAccountDiscoverySettingsChanges_1.TeamAccountDiscoverySettingsChanges,
    TeamChanges: teamChanges_1.TeamChanges,
    TeamCollaborationSettings: teamCollaborationSettings_1.TeamCollaborationSettings,
    TeamCollaborationSettingsChanges: teamCollaborationSettingsChanges_1.TeamCollaborationSettingsChanges,
    TeamCopyAccessLevelSettings: teamCopyAccessLevelSettings_1.TeamCopyAccessLevelSettings,
    TeamCopyAccessLevelSettingsChanges: teamCopyAccessLevelSettingsChanges_1.TeamCopyAccessLevelSettingsChanges,
    TeamInvitationSettings: teamInvitationSettings_1.TeamInvitationSettings,
    TeamInvitationSettingsChanges: teamInvitationSettingsChanges_1.TeamInvitationSettingsChanges,
    TeamMember: teamMember_1.TeamMember,
    TeamMemberChanges: teamMemberChanges_1.TeamMemberChanges,
    TeamMemberInvite: teamMemberInvite_1.TeamMemberInvite,
    TeamSettings: teamSettings_1.TeamSettings,
    TeamSettingsChanges: teamSettingsChanges_1.TeamSettingsChanges,
    TeamSharingPolicySettings: teamSharingPolicySettings_1.TeamSharingPolicySettings,
    TeamSharingPolicySettingsChanges: teamSharingPolicySettingsChanges_1.TeamSharingPolicySettingsChanges,
    TextCreateRequest: textCreateRequest_1.TextCreateRequest,
    TextData: textData_1.TextData,
    TextDataPlatformTags: textDataPlatformTags_1.TextDataPlatformTags,
    TextItem: textItem_1.TextItem,
    TextStyle: textStyle_1.TextStyle,
    TextUpdateRequest: textUpdateRequest_1.TextUpdateRequest,
    UpdateBoardsDataClassificationLabel: updateBoardsDataClassificationLabel_1.UpdateBoardsDataClassificationLabel,
    UpdateBoardsDataClassificationLabelRequest: updateBoardsDataClassificationLabelRequest_1.UpdateBoardsDataClassificationLabelRequest,
    UpdateTeamSettingsRequest: updateTeamSettingsRequest_1.UpdateTeamSettingsRequest,
    UserInfoShort: userInfoShort_1.UserInfoShort,
    WidgetData: widgetData_1.WidgetData,
    WidgetDataPlatformTags: widgetDataPlatformTags_1.WidgetDataPlatformTags,
    WidgetLinks: widgetLinks_1.WidgetLinks,
    WidgetLinksPlatformContainers: widgetLinksPlatformContainers_1.WidgetLinksPlatformContainers,
    WidthOnlyAdjustableGeometry: widthOnlyAdjustableGeometry_1.WidthOnlyAdjustableGeometry,
};
class ObjectSerializer {
    static findCorrectType(data, expectedType) {
        if (data == undefined) {
            return expectedType;
        }
        else if (primitives.indexOf(expectedType.toLowerCase()) !== -1) {
            return expectedType;
        }
        else if (expectedType === 'Date') {
            return expectedType;
        }
        else {
            if (enumsMap[expectedType]) {
                return expectedType;
            }
            if (!typeMap[expectedType]) {
                return expectedType; // w/e we don't know the type
            }
            // Check the discriminator
            let discriminatorProperty = typeMap[expectedType].discriminator;
            if (discriminatorProperty == null) {
                return expectedType; // the type does not have a discriminator. use it.
            }
            else {
                if (data[discriminatorProperty]) {
                    var discriminatorType = data[discriminatorProperty];
                    if (typeMap[discriminatorType]) {
                        return discriminatorType; // use the type given in the discriminator
                    }
                    else {
                        return expectedType; // discriminator did not map to a type
                    }
                }
                else {
                    return expectedType; // discriminator was not present (or an empty string)
                }
            }
        }
    }
    static serialize(data, type) {
        if (data == undefined) {
            return data;
        }
        else if (primitives.indexOf(type.toLowerCase()) !== -1) {
            return data;
        }
        else if (type.lastIndexOf('Array<', 0) === 0) {
            // string.startsWith pre es6
            let subType = type.replace('Array<', ''); // Array<Type> => Type>
            subType = subType.substring(0, subType.length - 1); // Type> => Type
            let transformedData = [];
            for (let index = 0; index < data.length; index++) {
                let datum = data[index];
                transformedData.push(ObjectSerializer.serialize(datum, subType));
            }
            return transformedData;
        }
        else if (type === 'Date') {
            return data.toISOString();
        }
        else {
            if (enumsMap[type]) {
                return data;
            }
            if (!typeMap[type]) {
                // in case we dont know the type
                return data;
            }
            // Get the actual type of this object
            type = this.findCorrectType(data, type);
            // get the map for the correct type.
            let attributeTypes = typeMap[type].getAttributeTypeMap();
            let instance = {};
            for (let index = 0; index < attributeTypes.length; index++) {
                let attributeType = attributeTypes[index];
                instance[attributeType.baseName] = ObjectSerializer.serialize(data[attributeType.name], attributeType.type);
            }
            return instance;
        }
    }
    static deserialize(data, type) {
        // polymorphism may change the actual type.
        type = ObjectSerializer.findCorrectType(data, type);
        if (data == undefined) {
            return data;
        }
        else if (primitives.indexOf(type.toLowerCase()) !== -1) {
            return data;
        }
        else if (type.lastIndexOf('Array<', 0) === 0) {
            // string.startsWith pre es6
            let subType = type.replace('Array<', ''); // Array<Type> => Type>
            subType = subType.substring(0, subType.length - 1); // Type> => Type
            let transformedData = [];
            for (let index = 0; index < data.length; index++) {
                let datum = data[index];
                transformedData.push(ObjectSerializer.deserialize(datum, subType));
            }
            return transformedData;
        }
        else if (type === 'Date') {
            return new Date(data);
        }
        else {
            if (enumsMap[type]) {
                // is Enum
                return data;
            }
            if (!typeMap[type]) {
                // dont know the type
                return data;
            }
            let instance = new typeMap[type]();
            let attributeTypes = typeMap[type].getAttributeTypeMap();
            for (let index = 0; index < attributeTypes.length; index++) {
                let attributeType = attributeTypes[index];
                instance[attributeType.name] = ObjectSerializer.deserialize(data[attributeType.baseName], attributeType.type);
            }
            return instance;
        }
    }
}
exports.ObjectSerializer = ObjectSerializer;
