// @mirohq/miro-api/dist/model/organizationMembersSearchResponse.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrganizationMembersSearchResponse = void 0;
/**
 * @internal
 * Response for query by cursor and filter parameters
 */
class OrganizationMembersSearchResponse {
    /** @ignore */
    static getAttributeTypeMap() {
        return OrganizationMembersSearchResponse.attributeTypeMap;
    }
}
exports.OrganizationMembersSearchResponse = OrganizationMembersSearchResponse;
/** @ignore */
OrganizationMembersSearchResponse.discriminator = undefined;
/** @ignore */
OrganizationMembersSearchResponse.attributeTypeMap = [
    {
        name: 'cursor',
        baseName: 'cursor',
        type: 'string',
    },
    {
        name: 'data',
        baseName: 'data',
        type: 'Array<OrganizationMember>',
    },
    {
        name: 'limit',
        baseName: 'limit',
        type: 'number',
    },
    {
        name: 'size',
        baseName: 'size',
        type: 'number',
    },
];
