// @mirohq/miro-api/dist/model/parentLinksEnvelopePlatformContainers.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ParentLinksEnvelopePlatformContainers = void 0;
/**
 * @internal
 * Contains information about the parent frame for the item.
 */
class ParentLinksEnvelopePlatformContainers {
    /** @ignore */
    static getAttributeTypeMap() {
        return ParentLinksEnvelopePlatformContainers.attributeTypeMap;
    }
}
exports.ParentLinksEnvelopePlatformContainers = ParentLinksEnvelopePlatformContainers;
/** @ignore */
ParentLinksEnvelopePlatformContainers.discriminator = undefined;
/** @ignore */
ParentLinksEnvelopePlatformContainers.attributeTypeMap = [
    {
        name: 'id',
        baseName: 'id',
        type: 'number',
    },
    {
        name: 'links',
        baseName: 'links',
        type: 'SelfLink',
    },
];
