// @mirohq/miro-api/dist/model/positionChange.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PositionChange = void 0;
/**
 * @internal
 * Contains location information about the item, such as its x coordinate, y coordinate, and theorigin of the x and y coordinates.
 */
class PositionChange {
    constructor() {
        /**
         * Area of the item that is referenced by its x and y coordinates. For example, an item with a center origin will have its x and y coordinates point to its center. The center point of the board has x: 0 and y: 0 coordinates. Currently, only one option is supported. Possible values: `center`
         */
        this['origin'] = PositionChange.OriginEnum.Center;
        /**
         * X-axis coordinate of the location of the item on the board. By default, all items have absolute positioning to the board, not the current viewport. The center point of the board has `x: 0` and `y: 0` coordinates.
         */
        this['x'] = 0;
        /**
         * Y-axis coordinate of the location of the item on the board. By default, all items have absolute positioning to the board, not the current viewport. The center point of the board has `x: 0` and `y: 0` coordinates.
         */
        this['y'] = 0;
    }
    /** @ignore */
    static getAttributeTypeMap() {
        return PositionChange.attributeTypeMap;
    }
}
exports.PositionChange = PositionChange;
/** @ignore */
PositionChange.discriminator = undefined;
/** @ignore */
PositionChange.attributeTypeMap = [
    {
        name: 'origin',
        baseName: 'origin',
        type: 'PositionChange.OriginEnum',
    },
    {
        name: 'x',
        baseName: 'x',
        type: 'number',
    },
    {
        name: 'y',
        baseName: 'y',
        type: 'number',
    },
];
(function (PositionChange) {
    PositionChange.OriginEnum = {
        Center: 'center',
    };
})(PositionChange = exports.PositionChange || (exports.PositionChange = {}));
