// @mirohq/miro-api/dist/model/stickyNoteData.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.StickyNoteData = void 0;
/**
 * @internal
 * Contains sticky note item data, such as the content or shape of the sticky note. For information about the JSON properties, see [Data](https://developers.miro.com/reference/data).
 */
class StickyNoteData {
    constructor() {
        /**
         * The actual text (content) that appears in the sticky note item.
         */
        this['content'] = 'Hello';
        /**
         * Defines the geometric shape of the sticky note and aspect ratio for its dimensions. Possible values: `square`, `rectangle`
         */
        this['shape'] = StickyNoteData.ShapeEnum.Square;
    }
    /** @ignore */
    static getAttributeTypeMap() {
        return StickyNoteData.attributeTypeMap;
    }
}
exports.StickyNoteData = StickyNoteData;
/** @ignore */
StickyNoteData.discriminator = undefined;
/** @ignore */
StickyNoteData.attributeTypeMap = [
    {
        name: 'content',
        baseName: 'content',
        type: 'string',
    },
    {
        name: 'shape',
        baseName: 'shape',
        type: 'StickyNoteData.ShapeEnum',
    },
];
(function (StickyNoteData) {
    StickyNoteData.ShapeEnum = {
        Square: 'square',
        Rectangle: 'rectangle',
    };
})(StickyNoteData = exports.StickyNoteData || (exports.StickyNoteData = {}));
