// @mirohq/miro-api/dist/model/teamMember.js
"use strict";
/**
 * Miro API
 * Miro API
 *
 * The version of the OpenAPI document: 0.1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TeamMember = void 0;
class TeamMember {
    /** @ignore */
    static getAttributeTypeMap() {
        return TeamMember.attributeTypeMap;
    }
}
exports.TeamMember = TeamMember;
/** @ignore */
TeamMember.discriminator = undefined;
/** @ignore */
TeamMember.attributeTypeMap = [
    {
        name: 'createdAt',
        baseName: 'createdAt',
        type: 'Date',
    },
    {
        name: 'createdByUserId',
        baseName: 'createdByUserId',
        type: 'number',
    },
    {
        name: 'memberId',
        baseName: 'memberId',
        type: 'number',
    },
    {
        name: 'modifiedAt',
        baseName: 'modifiedAt',
        type: 'Date',
    },
    {
        name: 'modifiedByUserId',
        baseName: 'modifiedByUserId',
        type: 'number',
    },
    {
        name: 'teamId',
        baseName: 'teamId',
        type: 'number',
    },
    {
        name: 'type',
        baseName: 'type',
        type: 'string',
    },
    {
        name: 'userRole',
        baseName: 'userRole',
        type: 'TeamMember.UserRoleEnum',
    },
];
(function (TeamMember) {
    TeamMember.UserRoleEnum = {
        NonTeam: 'non_team',
        Member: 'member',
        Admin: 'admin',
        TeamGuest: 'team_guest',
    };
})(TeamMember = exports.TeamMember || (exports.TeamMember = {}));
